anomdetect/src/test/java/com/expedia/adaptivealerting/anomdetect/forecast/point/algo/sma/SmaPointForcasterTest.java

```diff
@@ -0,0 +1,117 @@
+package com.expedia.adaptivealerting.anomdetect.forecast.point.algo.sma;
+
+import java.io.InputStreamReader;
+import java.util.Arrays;
+import java.util.List;
+import java.util.function.Function;
+
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import com.expedia.metrics.MetricData;
+import com.expedia.metrics.MetricDefinition;
+import com.opencsv.bean.CsvToBeanBuilder;
+
+import lombok.val;
+
+import static org.junit.Assert.assertEquals;
+
+public class SmaPointForcasterTest {
+    private static final double TOLERANCE = 0.001;
+
+    private static List<SmaPointForecasterTestRow> testRows;
+
+    @BeforeClass
+    public static void setUpClass() {
+        loadSampleData();
+    }
+
+    @Test
+    public void testWithInitialValues() {
+        SmaPointForecasterParams params = new SmaPointForecasterParams()
+            .setLookBackPeriod(3)
+            .setInitialPeriodOfValues(Arrays.asList(1.0, 2.0, 3.0));
+        SmaPointForecaster sma3 = new SmaPointForecaster(params);
+
+        // [1.0, 2.0, 3.0]; mean=2.0
+        assertEquals(2.0, sma3.getMean(), TOLERANCE);
+        // [2.0, 3.0, 4.0]; mean=3.0
+        assertEquals(3.0, sma3.forecast(createMetricData(4.0)).getValue(), TOLERANCE);
+        // [3.0, 4.0, 5.0]; mean=4.0
+        assertEquals(4.0, sma3.forecast(createMetricData(5.0)).getValue(), TOLERANCE);
+    }
+
+    @Test
+    public void testNoInitialValues() {
+        SmaPointForecasterParams params = new SmaPointForecasterParams().setLookBackPeriod(3);
+        SmaPointForecaster sma3 = new SmaPointForecaster(params);
+
+        // [2.0]; mean=2.0
+        assertEquals(2.0, sma3.forecast(createMetricData(2.0)).getValue(), TOLERANCE);
+        // [2.0, 4.0]; mean=3.0
+        assertEquals(3.0, sma3.forecast(createMetricData(4.0)).getValue(), TOLERANCE);
+        // [2.0, 4.0, 6.0]; mean=4.0
+        assertEquals(4.0, sma3.forecast(createMetricData(6.0)).getValue(), TOLERANCE);
+        // [4.0, 6.0, 8.0]; mean=6.0
+        assertEquals(6.0, sma3.forecast(createMetricData(8.0)).getValue(), TOLERANCE);
+    }
+
+    @Test
+    public void testAgainstSampleData() {
```

Comment:
Our sample data in the CSV has pre-calculated values for SMA(1), SMA(3), SMA(9), and SMA(21). This one method will test all of these forecasters at the same time in a single iteration over the dataset.